docroot/modules/custom/foia_webform/tests/src/Kernel/FoiaWebformApiKernelTestBase.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Drupal\Tests\foia_webform\Kernel;
+
+use Drupal\taxonomy\Entity\Vocabulary;
+use Drupal\taxonomy\Entity\Term;
+use Drupal\webform\Entity\Webform;
+use Drupal\node\Entity\NodeType;
+use Drupal\node\Entity\Node;
+
+/**
+ * Base Class for FOIA Webform Kernel tests that use the API.
+ */
+abstract class FoiaWebformApiKernelTestBase extends FoiaWebformKernelTestBase {
+
+  use ReflectionTrait;
+  use FieldInstallTrait;
+
+  /**
+   * Test agency.
+   *
+   * @var \Drupal\taxonomy\Entity\Term
+   */
+  protected $agency;
+
+  /**
+   * Test webform to submit against.
+   *
+   * @var \Drupal\webform\WebformInterface
+   */
+  protected $webform;
+
+  /**
+   * Test webform submission.
+   *
+   * @var \Drupal\webform\WebformSubmissionInterface
+   */
+  protected $webformSubmission;
+
+  /**
+   * Test agency component we're submitting to.
+   *
+   * @var \Drupal\node\NodeInterface
+   */
+  protected $agencyComponent;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  protected static $modules = [
+    'foia_request',
+    'link',
+    'options',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->installSchema('webform', ['webform']);
+    $this->installConfig(['webform', 'webform_template', 'foia_webform']);
```

Comment:
I know we don't use `webform`, `webform_template`, and `foia_webform` in _every_ class, but I think we could move these to the `FoiaWebformKernelTestBase`, since it wouldn't be farfetched to assume that a class that extends the base that isn't testing API-related items would need some of these items.